client/ayon_harmony/plugins/create/create_workfile.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+"""Creator plugin for creating workfiles."""
+import ayon_api
+
+from ayon_core.pipeline import CreatedInstance, AutoCreator
+from ayon_harmony.api import plugin, lib
+
+
+class CreateWorkfile(plugin.HarmonyCreatorBase, AutoCreator):
+    """Workfile auto-creator."""
+    identifier = "io.ayon.creators.harmony.workfile"
+    label = "Workfile"
+    product_type = "workfile"
+    icon = "fa5.file"
+
+    default_variant = "Main"
+
+    # Placeholder node name for where we store the workfile data.
+    # This does not create an actual Harmony node, but just uses this name
+    # as key in the AYON Harmony scene data.
+    _node_name = "__workfile__"
+
+    def create(self):
+
+        variant = self.default_variant
+        current_instance = next(
+            (
+                instance for instance in self.create_context.instances
+                if instance.creator_identifier == self.identifier
+            ), None)
+
+        project_name = self.project_name
+        folder_path = self.create_context.get_current_folder_path()
```

Comment:
We can use `self.create_context.get_current_folder_entity()`, and for task entity too, which might make it faster as it already might be pre-fetched.

Reply:
Thanks - this was mostly copy pasted from another host integration so it may be relevant in others too. I'll refactor this one.

Reply:
Implemented with https://github.com/ynput/ayon-harmony/pull/17/commits/20939f5b6ada425a545bc4988e5232501be505a2